Sources/LocationMessageCell.swift

```diff
@@ -69,11 +70,33 @@ open class LocationMessageCell: MessageCollectionViewCell<UIImageView> {
 
         let snapShotter = MKMapSnapshotter(options: snapshotOptions)
         snapShotter.start { (snapshot, error) in
-            self.activityIndicator.stopAnimating()
-            if error == nil {
-                self.messageContentView.image = snapshot?.image
+            defer {
+                self.activityIndicator.stopAnimating()
             }
+            guard let snapshot = snapshot, error == nil else {
+                //show an error image?
+                return
+            }
+
+            UIGraphicsBeginImageContextWithOptions(snapshotOptions.size, true, 0)
+
+            snapshot.image.draw(at: .zero)
+
+            var point = snapshot.point(for: location.coordinate)
+            //Move point to reflect annotation anchor
+            point.x -= annotationView.bounds.size.width / 2
+            point.y -= annotationView.bounds.size.height / 2
+            point.x += annotationView.centerOffset.x
+            point.y += annotationView.centerOffset.y
+
+            annotationView.image?.draw(at: point)
```

Comment:
My only question is what if the user is not using an image for the MKAnnotationView? It seems like we're not using MKAnnotationView at all really. It's just wrapping up our UIImage. Am I making sense?

Reply:
@etoledom Isn't there some way to draw the `MKAnnotationView` as a whole?

Reply:
What I'm trying to say is what if I want to style my `MKAnnotationView` using the properties of `UIView` and not just a `UIImage`

Reply:
@SD10 I see what you mean.

We can actually draw the annotationView's layer with:
`annotationView.layer.render(in: context)`
I tested it and seems to work fine.

For some reason, every tutorial and example I have found uses the image drawing thing, but the Apple documentation on `MKAnnotationView` says that is allowed also to draw the pin overriding `draw(_:)`, so in that case probably will be the best, as you said, to draw the whole view instead of just the image for maximum flexibility.

I will test it a bit more, and PR with this little drawing change.

Reply:
Works on `MKAnnotationView`, `MKMarkerAnnotationView`, custom subclasses of `MKAnnotationView`, but it doesn't want to work on `MKPinAnnotationView` (our default) 😅